spec/models/recipe_spec.rb

```diff
@@ -0,0 +1,121 @@
+# /spec/models/recipe_spec.rb
+require 'rails_helper'
+
+RSpec.describe Recipe, type: :model do
+  let(:recipe) { Recipe.new } # Define a recipe object
+
+  it 'is valid with valid attributes' do
+    recipe.valid? # Validate the recipe without saving it
+    expect(recipe).to be_valid
+  end
+
+  it 'is not valid without a name' do
+    recipe.name = nil
+    recipe.valid?
+    expect(recipe).not_to be_valid
+    expect(recipe.errors[:name]).to include("can't be blank")
+  end
+
+  it 'is not valid with a name less than 3 characters' do
+    recipe.name = 'ab'
+    recipe.valid?
+    expect(recipe).not_to be_valid
+    expect(recipe.errors[:name]).to include('is too short (minimum is 3 characters)')
+  end
+
+  it 'is not valid with a name more than 50 characters' do
+    recipe.name = 'a' * 51
+    recipe.valid?
```

Comment:
- You have done well testing your application, but some of your tests are failing . Please make sure all tests pass before requesting another review. Please fix this 👍 👍 
![dasrgv](https://github.com/AbdelazizSoliman/Recipe_app/assets/73607512/bd816db6-292e-47e6-8bf1-b0332f43eba0)
- Good job done so far, but according to the project requirements, you should write both unit and integration test for your application. Please make sure you include integration test for your app 👍 👍